File: make_dmg_bg.py
import Cocoa

width = 600
height = 400

image = Cocoa.NSImage.alloc().initWithSize_(Cocoa.NSMakeSize(width, height))
image.lockFocus()

# Draw background (gradient from top to bottom)
gradient = Cocoa.NSGradient.alloc().initWithStartingColor_endingColor_(
    Cocoa.NSColor.colorWithCalibratedRed_green_blue_alpha_(0.96, 0.96, 0.98, 1.0),
    Cocoa.NSColor.colorWithCalibratedRed_green_blue_alpha_(0.90, 0.92, 0.95, 1.0)
)
gradient.drawInRect_angle_(Cocoa.NSMakeRect(0, 0, width, height), -90)

# Draw Title
title = "가마지기"
titleFont = Cocoa.NSFont.systemFontOfSize_weight_(48, Cocoa.NSFontWeightBold)
titleAttrs = {Cocoa.NSFontAttributeName: titleFont, Cocoa.NSForegroundColorAttributeName: Cocoa.NSColor.colorWithCalibratedWhite_alpha_(0.2, 1.0)}
Cocoa.NSString.stringWithString_(title).drawAtPoint_withAttributes_(Cocoa.NSMakePoint(40, 310), titleAttrs)

# Draw Subtitle
subtitle = "보카보카 250BT 커피 로스터 프로파일링 앱"
subFont = Cocoa.NSFont.systemFontOfSize_weight_(18, Cocoa.NSFontWeightMedium)
subAttrs = {Cocoa.NSFontAttributeName: subFont, Cocoa.NSForegroundColorAttributeName: Cocoa.NSColor.colorWithCalibratedWhite_alpha_(0.4, 1.0)}
Cocoa.NSString.stringWithString_(subtitle).drawAtPoint_withAttributes_(Cocoa.NSMakePoint(42, 280), subAttrs)

# Draw arrow or indicator
arrowFont = Cocoa.NSFont.systemFontOfSize_weight_(40, Cocoa.NSFontWeightLight)
arrowAttrs = {Cocoa.NSFontAttributeName: arrowFont, Cocoa.NSForegroundColorAttributeName: Cocoa.NSColor.colorWithCalibratedWhite_alpha_(0.6, 1.0)}
Cocoa.NSString.stringWithString_("➜").drawAtPoint_withAttributes_(Cocoa.NSMakePoint(280, 150), arrowAttrs)

image.unlockFocus()

bitmap = Cocoa.NSBitmapImageRep.alloc().initWithData_(image.TIFFRepresentation())
pngData = bitmap.representationUsingType_properties_(Cocoa.NSPNGFileType, None)
pngData.writeToFile_atomically_("dmg_background.png", True)
print("Background generated: dmg_background.png")

File: make_dmg_bg.swift
import Cocoa

let width: CGFloat = 600
let height: CGFloat = 400

let image = NSImage(size: NSSize(width: width, height: height))
image.lockFocus()

// Draw background
if let gradient = NSGradient(starting: NSColor(calibratedRed: 0.96, green: 0.96, blue: 0.98, alpha: 1.0),
                             ending: NSColor(calibratedRed: 0.90, green: 0.92, blue: 0.95, alpha: 1.0)) {
    gradient.draw(in: NSRect(x: 0, y: 0, width: width, height: height), angle: -90)
}

// Draw Title
let title = "보카보아"
let titleFont = NSFont.systemFont(ofSize: 48, weight: .bold)
let titleAttrs: [NSAttributedString.Key: Any] = [
    .font: titleFont,
    .foregroundColor: NSColor(calibratedWhite: 0.2, alpha: 1.0)
]
(title as NSString).draw(at: NSPoint(x: 40, y: 310), withAttributes: titleAttrs)

// Draw Subtitle
let subtitle = "보카보카250BT 커피로스터 로스팅 프로파일러"
let subFont = NSFont.systemFont(ofSize: 18, weight: .medium)
let subAttrs: [NSAttributedString.Key: Any] = [
    .font: subFont,
    .foregroundColor: NSColor(calibratedWhite: 0.4, alpha: 1.0)
]
(subtitle as NSString).draw(at: NSPoint(x: 42, y: 280), withAttributes: subAttrs)

// Draw arrow
let arrowFont = NSFont.systemFont(ofSize: 40, weight: .light)
let arrowAttrs: [NSAttributedString.Key: Any] = [
    .font: arrowFont,
    .foregroundColor: NSColor(calibratedWhite: 0.6, alpha: 1.0)
]
("➜" as NSString).draw(at: NSPoint(x: 280, y: 160), withAttributes: arrowAttrs)

image.unlockFocus()

if let tiffData = image.tiffRepresentation,
   let bitmap = NSBitmapImageRep(data: tiffData),
   let pngData = bitmap.representation(using: .png, properties: [:]) {
    try? pngData.write(to: URL(fileURLWithPath: "dmg_background.png"))
    print("Background generated: dmg_background.png")
}
